Build failure: nomad

### Steps To Reproduce

Steps to reproduce the behavior:
1. nix-env -i nomad-1.5.8

### Build log

```
$ nix-env -i nomad-1.5.8
replacing old 'nomad-1.4.6'
installing 'nomad-1.5.8'
this derivation will be built:
  /nix/store/7br9jb98m49hlpb304cvc20xq93d6z1m-nomad-1.5.8.drv
building '/nix/store/7br9jb98m49hlpb304cvc20xq93d6z1m-nomad-1.5.8.drv'...
unpacking sources
unpacking source archive /nix/store/d9v6ybz2414mh0siw12y0vblxylh69vx-source
source root is source
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
building
Building subPackage ./.
running tests
panic: failed to find nomad binary: exec: "nomad": executable file not found in $PATH

goroutine 1 [running]:
github.com/hashicorp/nomad/helper/subproc.init.0()
        /private/tmp/nix-build-nomad-1.5.8.drv-0/source/helper/subproc/self.go:25 +0x1e4
FAIL    github.com/hashicorp/nomad      0.440s
FAIL
error: builder for '/nix/store/7br9jb98m49hlpb304cvc20xq93d6z1m-nomad-1.5.8.drv' failed with exit code 1;
       last 10 log lines:
       > building
       > Building subPackage ./.
       > running tests
       > panic: failed to find nomad binary: exec: "nomad": executable file not found in $PATH
       >
       > goroutine 1 [running]:
       > github.com/hashicorp/nomad/helper/subproc.init.0()
       >      /private/tmp/nix-build-nomad-1.5.8.drv-0/source/helper/subproc/self.go:25 +0x1e4
       > FAIL  github.com/hashicorp/nomad      0.440s
       > FAIL
       For full logs, run 'nix-store -l /nix/store/7br9jb98m49hlpb304cvc20xq93d6z1m-nomad-1.5.8.drv'

$ nix-store -l /nix/store/7br9jb98m49hlpb304cvc20xq93d6z1m-nomad-1.5.8.drv
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/d9v6ybz2414mh0siw12y0vblxylh69vx-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Building subPackage ./.
@nix { "action": "setPhase", "phase": "checkPhase" }
running tests
panic: failed to find nomad binary: exec: "nomad": executable file not found in $PATH

goroutine 1 [running]:
github.com/hashicorp/nomad/helper/subproc.init.0()
        /private/tmp/nix-build-nomad-1.5.8.drv-0/source/helper/subproc/self.go:25 +0x1e4
FAIL    github.com/hashicorp/nomad      0.440s
FAIL
```

### Additional context

Build fails for both nomad-1.5.8 and nomad-1.6.1. nomad-1.4.6 installs without build.

### Notify maintainers

@rushmorem @pradeepchhetri @endocrimes @maxeaubrey @techknowlogick

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.5.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.17.0`
 - channels(michael): `""`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

--- a/pkgs/applications/networking/cluster/nomad/default.nix
+++ b/pkgs/applications/networking/cluster/nomad/default.nix
@@ -82,7 +82,7 @@ rec {
     vendorSha256 = "sha256-y3WiQuoQn6SdwTgtPWuB6EBtsJC+YleQPzownZQNkno=";
     passthru.tests.nomad = nixosTests.nomad;
     preCheck = ''
-      export PATH="$PATH:/build/go/bin"
+      export PATH="$PATH:$NIX_BUILD_TOP/go/bin"
     '';
   };
 
@@ -93,7 +93,7 @@ rec {
     vendorSha256 = "sha256-Y3O7ADzZPlLWFbXSYBcI6b5MAhMD0UnkhQxO9VJMpOY=";
     passthru.tests.nomad = nixosTests.nomad;
     preCheck = ''
-      export PATH="$PATH:/build/go/bin"
+      export PATH="$PATH:$NIX_BUILD_TOP/go/bin"
     '';
   };
 }

